tmol/score/constraint/constraint_energy_term.py

```diff
@@ -0,0 +1,142 @@
+import torch
+import numpy
+import math
+
+from ..energy_term import EnergyTerm
+
+from tmol.database import ParameterDatabase
+from tmol.score.constraint.constraint_whole_pose_module import (
+    ConstraintWholePoseScoringModule,
+)
+from tmol.score.constraint.potentials.compiled import get_torsion_angle
+
+from tmol.chemical.restypes import RefinedResidueType
+from tmol.pose.packed_block_types import PackedBlockTypes
+from tmol.pose.pose_stack import PoseStack
+
+
+class ConstraintEnergyTerm(EnergyTerm):
+    device: torch.device  # = attr.ib()
+
+    def __init__(self, param_db: ParameterDatabase, device: torch.device):
+        super(ConstraintEnergyTerm, self).__init__(param_db=param_db, device=device)
+
+        self.device = device
+
+    @classmethod
+    def score_types(cls):
+        import tmol.score.terms.constraint_creator
+
+        return tmol.score.terms.constraint_creator.ConstraintTermCreator.score_types()
+
+    def n_bodies(self):
+        return 1
```

Comment:
Not totally sure what to do here, since this could involve up to 4 'bodies'. But as I understand it, this is mostly for dispatch reasons, which isn't a concern in this case.